src/types.d.ts

```diff
@@ -116,6 +122,8 @@ interface Rule {
   // Whether to pass all the events in the scope to the matcher. If false, only the scope event
   // is passed to the matcher, and the rule should traverse the scope as needed.
   enumerateScope: boolean;
+  impactDomain?: ImpactDomain;
+  references?: Record<string, URL>;
```

Comment:
Not sure if it's not better as a simple string than `URL`. It's purely descriptive and not manipulated anywhere, it's going to be re-stringified again before using, plus it's a bit clunky to have to import *URL* and create the object every time.

Reply:
I mean yes, but, I feel better knowing that the URLs are valid.

Reply:
Ok, sounds fair